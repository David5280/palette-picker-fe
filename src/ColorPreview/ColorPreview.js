import React, { Component } from 'react'
import ColorScheme from 'color-scheme'
import Palette from '../Palette/Palette';
import { serverCall } from '../fetchCalls/fetchCalls';

export class ColorPreview extends Component {
  state ={
    edited: false,
    color: ''
  }
  
  randomColor = () => {
    const randomHue = (Math.random() * (360 - 1) + 1)
    const colorScheme = new ColorScheme()
    colorScheme.from_hue(randomHue)
    const color = `#${colorScheme.colors()[0]}`
    this.setState({ edited: true, color: color })
  }

  patchColor = async (paletteID, dbKey) => {
    console.log(paletteID, dbKey)
    const patchRes = await serverCall(`palettes/${paletteID}`, "PATCH", {[dbKey]:this.state.color})
    // console.log(await patchRes)
  }

  render() {
    const { color, paletteID, dbKey } = this.props;
    return (
      <div 
      className='single-color-preview' 
      style={{ backgroundColor: `${this.state.color || color}`, width:'20%', marginBottom:'10px', height:50, verticalAlign:'center', display: 'flex', alignItems: 'center', justifyContent: 'center'}}
      >
        <button
        onClick={() => this.randomColor()}
        >
          Randomize Color
        </button>
      
        <p style={{ backgroundColor: 'rgba(255, 255, 255, 0.3)', width: '50%' }}>
        {this.state.color || color}
        {this.state.edited && <button
        onClick={() => this.patchColor(paletteID, dbKey)} 
        >Save to Palette!</button>}
        </p>
      </div>
    )
  }
}

export default ColorPreview
