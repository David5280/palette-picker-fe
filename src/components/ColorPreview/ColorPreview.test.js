import React from 'react';
import ColorPreview from './ColorPreview';
import { shallow } from 'enzyme';

describe('ColorPreview', () => {
  let mockColorObj;
  let wrapper;
  beforeEach(() => {
    mockColorObj = {
      color: '#FFFEEE',
      paletteID: 4,
      dbKey: 'AYEE',
      }
    wrapper = shallow(
      <ColorPreview 
        color={mockColorObj.color}
        paletteID={mockColorObj.paletteID}
        dbKey={mockColorObj.dbKey}
        hexcode={'#ddbbcc'}
        index={1}
        handleLock={jest.fn()}
        patchProject={jest.fn()}
        icon={'/newIcon.png'}
      />
    )
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should change state when randomColor fires', () => {
    expect(wrapper.state('color')).toEqual('')    
    wrapper.instance().randomColor()
    expect(wrapper.state('edited')).toEqual(true)
    expect(wrapper.state('color')).not.toEqual('')

  });
  it('should change state.edited from true to false when patchColor fires', () => {
    wrapper.instance().randomColor()
    expect(wrapper.state('edited')).toEqual(true)
    wrapper.instance().patchColor()
    expect(wrapper.state('edited')).toEqual(false)
  });
  it('should fire randomColor() when randomize-icon is clicked', () => {
    wrapper.instance().randomColor = jest.fn();
    wrapper.find('.randomize-icon').simulate('click');
    expect(wrapper.instance().randomColor).toHaveBeenCalled();
  });
  it('should fire patchColor() when save-icon is clicked', () => {
    wrapper.instance().patchColor = jest.fn();
    wrapper.find('.randomize-icon').simulate('click');
    wrapper.find('.save-icon').simulate('click');
    expect(wrapper.instance().patchColor).toHaveBeenCalled();
  });
});