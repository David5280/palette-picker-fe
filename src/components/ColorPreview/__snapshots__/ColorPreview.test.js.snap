// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPreview should match the snapshot 1`] = `ShallowWrapper {}`;
